python_scripts/repeatable/strip_whitespace.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+# This script works through a csv of offending content and removes whitespace from the specified field(s).
+# For help on available arguments and options:
+# `python repeatable/strip_whitespace.py -h`.
+import argparse
+import os
+import sys
+
+from asnake.aspace import ASpace
+from dotenv import load_dotenv, find_dotenv
+from loguru import logger
+from pathlib import Path
+from python_scripts.utilities import read_csv, record_error
+
+# Logging
+logger.remove()
+log_name = __file__.rsplit('/',1)[1].replace('.py', '')+'_{time:YYYY-MM-DD}'
+log_path = Path(f'./logs/{log_name}.log')
+logger.add(str(log_path), format="{time}-{level}: {message}")
+
+# Find  and load environment-specific .env file
+env_file = find_dotenv(f'.env.{os.getenv("ENV", "dev")}')
+load_dotenv(env_file)
+
+# Define functions
+def archival_objects(repo): return repo.archival_objects
+def digital_objects(repo): return repo.digital_objects
+def resources(repo): return repo.resources
+
+# Create a dictionary to map arguments to functions
+func_dict = {
+    "archival_objects": archival_objects,
+    "digital_objects": digital_objects,
+    "resources": resources
+}
+
+def parseArguments():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument("csvPath", help="path to csv input file", type=str)
+    parser.add_argument("-dR", "--dry-run", help="dry run?", action='store_true')
+    parser.add_argument("-t", "--type", help="type of ArchivesSpace object", choices=func_dict.keys(), required='true')
+    parser.add_argument("--version", action="version", version='%(prog)s - Version 1.0')
+
+    return parser.parse_args()
+
+def strip_whitespace(json_object, field_1, field_2):
+    """
+    Will strip whitespace from a single field when supplied with an ArchivesSpace json object and
+    the field to update.  If the targeted field is nested, may supply the parent field in field_1
+    and field to act upon in field_2.
+
+    Args:
+        json_object (dict): json object from ArchivesSpace
+        field_1 (str): the key of the field to strip whitespace from, or, if the targeted field is a
+        nested field, the parent of the target field
+        field_2 (str): the key of the nested field to strip whitespace from. Will be blank if the
+        targeted field is not nested in a subrecord.
+    """
+    updated_object = {}
+    if field_1 in json_object:
+        object_1 = json_object[field_1]
+        if field_2:
+            for i, item in list(enumerate(object_1)):
```

Comment:
I haven't seen enumerate() before, very nifty! So it returns the index of the nested object field and the value?

Reply:
Yep, it returns the index (which I've set to `i`) and the value (here defined to `item`), starting at index 0 (unless you set a different start).  So, a handy way to account for cases where nested objects might be repeated (e.g. a digital object can have multiple file_versions that each of whitespace-containing file_uris).